Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/ScaleHandles.cs

```diff
@@ -63,44 +70,116 @@ internal void Create(ref Vector3[] boundsCorners, Transform parent, bool drawMan
-                    cornerVisual.transform.parent = visualsScale.transform;
-                    cornerVisual.transform.localPosition = Vector3.zero;
-                    GameObject.Destroy(cornerVisual.GetComponent<BoxCollider>());
-                }
-                else
-                {
-                    cornerVisual = GameObject.Instantiate(prefabType, visualsScale.transform);
-                }
+                Bounds visualBounds = CreateVisual(visualsScale, isFlattened);
+
+                var invScale = config.HandleSize / visualBounds.size.x;
+                VisualUtils.AddComponentsToAffordance(corner, new Bounds(visualBounds.center * invScale, visualBounds.size * invScale), 
+                    HandlePrefabCollider.Box, CursorContextInfo.CursorAction.Scale, config.ColliderPadding, parent, config.DrawTetherWhenManipulating);
+                handles.Add(corner.transform);       
+            }
 
-                if (isFlattened)
+            VisualUtils.HandleIgnoreCollider(config.HandlesIgnoreCollider, handles);
+            objectsChangedEvent.Invoke(this);
+        }
+
+        protected override void RecreateVisuals()
+        {
+            for (int i = 0; i < handles.Count; ++i)
+            {
+                // get parent of visual
+                Transform visualsScaleParent = handles[i].Find("visualsScale");
+                if (visualsScaleParent)
                 {
-                    // Rotate 2D slate handle asset for proper orientation
-                    cornerVisual.transform.Rotate(0, 0, -90);
+                    // get old child and remove it
+                    Transform obsoleteChild = visualsScaleParent.Find("visuals");
+                    if (obsoleteChild)
+                    {
+                        obsoleteChild.parent = null;
+                        Object.Destroy(obsoleteChild.gameObject);
+                    }
+                    else
+                    {
+                        Debug.LogError("couldn't find corner visual on recreating visuals");
+                    }
+
+                    // create new visual
+                    Bounds cornerBounds = CreateVisual(visualsScaleParent.gameObject, areHandlesFlattened);
+
+                    // update handle collider bounds
+                    UpdateColliderBounds(handles[i], cornerBounds.size);
                 }
+            }
 
-                cornerVisual.name = "visuals";
+            objectsChangedEvent.Invoke(this);
+        }
 
-                // this is the size of the corner visuals
-                var cornerbounds = VisualUtils.GetMaxBounds(cornerVisual);
-                float maxDim = Mathf.Max(Mathf.Max(cornerbounds.size.x, cornerbounds.size.y), cornerbounds.size.z);
-                cornerbounds.size = maxDim * Vector3.one;
+        protected override void UpdateColliderBounds(Transform handle, Vector3 visualSize)
+        {
+            var invScale = config.HandleSize / visualSize.x;
```

Comment:
possible divide by zero here also.

Reply:
made sure this is fixed in rotation handles as well. thanks for pointing this out!